### PR TITLE
Split ServerConfig and ClientConfig across multiple files

### PR DESCRIPTION
**Pull request by mjcarroll**
A single header was containing both classes, this just splits those up.

### CODE REVIEW

File: include/gz/fuel_tools/ServerConfig.hh [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+ * limitations under the License.
+ *
+*/
+
+#ifndef GZ_FUEL_TOOLS_SERVERCONFIG_HH_
+#define GZ_FUEL_TOOLS_SERVERCONFIG_HH_
+
+#include <memory>
+#include <string>
+#include <vector>
```
Comment (ahcorde) [inline include/gz/fuel_tools/ServerConfig.hh:23]: where is vector used?

File: src/ServerConfig.cc [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+ * limitations under the License.
+ *
+*/
+
+#include <yaml.h>
+#include <cstdio>
+#include <sstream>
+#include <stack>
+#include <string>
+#include <vector>
```
Comment (ahcorde) [inline src/ServerConfig.cc:23]: where is vector used?

File: src/ServerConfig_TEST.cc [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  EXPECT_TRUE(config.ApiKey().empty());
+
+  config.SetApiKey("my_api_key");
+  EXPECT_EQ("my_api_key", config.ApiKey());
+
+  config.SetApiKey("my_other_api_key");
+  EXPECT_EQ("my_other_api_key", config.ApiKey());
+}
+
+
```
Comment (ahcorde) [inline src/ServerConfig_TEST.cc:81]:
```suggestion
}

```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: include/gz/fuel_tools/ServerConfig.hh (hunk 1 above)
Comment (mjcarroll) [reply include/gz/fuel_tools/ServerConfig.hh:23]: removed

File: src/ServerConfig.cc (hunk 2 above)
Comment (mjcarroll) [reply src/ServerConfig.cc:23]: removed

**Review by ahcorde — APPROVED** [review]

**Comment by mjcarroll** [comment]
Windows failures will be addressed with forward ports of #376

**Pull request merged**